pyccl/nl_pt/power.py

```diff
@@ -0,0 +1,583 @@
+            bs2 (array_like): tidal bias for the second tracer
+                being correlated at the same set of input redshifts.
+            sub_lowk (bool): if True, the small-scale white noise
+                contribution will be subtracted.
+
+        Returns:
+            array_like: 2D array of shape `(N_k, N_z)`, where `N_k` \
+                is the size of this object's `ks` attribute, and \
+                `N_z` is the size of the input redshift-dependent \
+                biases and growth factor.
+        """
+        Pd1d2 = g4[None, :] * self.dd_bias[2][:, None]
+        Pd2d2 = g4[None, :] * self.dd_bias[3][:, None]
+        Pd1s2 = g4[None, :] * self.dd_bias[4][:, None]
+        Pd2s2 = g4[None, :] * self.dd_bias[5][:, None]
+        Ps2s2 = g4[None, :] * self.dd_bias[6][:, None]
+
+        s4 = 0.
+        if sub_lowk:
+            s4 = g4 * self.dd_bias[7]
+            s4 = s4[None, :]
+
+        pgg = ((b11*b12)[None, :] * Pd1d1 +
+               0.5*(b11*b22 + b12*b21)[None, :] * Pd1d2 +
+               0.25*(b21*b22)[None, :] * (Pd2d2 - 2.*s4) +
+               0.5*(b11*bs2 + b12*bs1)[None, :] * Pd1s2 +
+               0.25*(b21*bs2 + b22*bs1)[None, :] * (Pd2s2 - (4./3.)*s4) +
+               0.25*(bs1*bs2)[None, :] * (Ps2s2 - (8./9.)*s4))
+        return pgg
+
+    def get_pgi(self, Pd1d1, g4, b1, b2, bs, c1, c2, cd):
+        """ Get the number counts - IA cross-spectrum at the
+        internal set of wavenumbers (given by this object's
+        `ks` attribute) and a number of redshift values.
+
+        Args:
+            Pd1d1 (array_like): 1-loop matter power spectrum at the
+                wavenumber values given by this object's `ks` list.
+            g4 (array_like): fourth power of the growth factor at
+                a number of redshifts.
+            b1 (array_like): 1-st order bias for the number counts
+                being correlated at the same set of input redshifts.
+            b2 (array_like): 2-nd order bias for the number counts
+                being correlated at the same set of input redshifts.
+            bs (array_like): tidal bias for the number counts
+                being correlated at the same set of input redshifts.
+            c1 (array_like): 1-st order bias for the IA tracer
+                being correlated at the same set of input redshifts.
+            c2 (array_like): 2-nd order bias for the IA tracer
+                being correlated at the same set of input redshifts.
+            cd (array_like): overdensity bias for the IA tracer
+                being correlated at the same set of input redshifts.
+
+        Returns:
+            array_like: 2D array of shape `(N_k, N_z)`, where `N_k` \
+                is the size of this object's `ks` attribute, and \
+                `N_z` is the size of the input redshift-dependent \
+                biases and growth factor.
+        """
+        warnings.warn(
```

Comment:
This warning is a bit much. I'd say we put this in the docstring and not bother people.

Reply:
Done